### ISSUE TITLE
canvas.node: invalid ELF header

### ISSUE THREAD

**Issue by pruteanualex** [Automattic/node-canvas#2258]
## Issue or Feature
- [ ] If this is an issue with installation, I have read the [troubleshooting guide](https://github.com/Automattic/node-canvas/issues/1511).

## Steps to Reproduce
```js
const GIFEncoder = require('gifencoder');
const { createCanvas, loadImage} = require('canvas');;
const canvas = createCanvas(width , height );
const ctx = canvas.getContext('2d');

// Encode gif settings
    const encoder = new GIFEncoder(width , height );
    encoder.createReadStream().pipe(fs.createWriteStream(tempPath+'/user_gif_'+clientID+'/gif/giff_generated.gif'));
    encoder.start();
    encoder.setRepeat(0);
    encoder.setDelay(gif_delay);
    encoder.setQuality(gif_quality);
    
    // read image
    const imgList = fs.readdirSync(tempPath+'/user_gif_'+clientID+'/converted');
    imgList.forEach(async (f, i) => {
        let imagePath = `${tempPath}/user_gif_${clientID}/converted/${f}`;
        const image =  await loadImage(imagePath);
        if(image){
            ctx.drawImage(image, 0, 0, image.width, image.height, 0, 0, canvas.width, canvas.height);
            encoder.addFrame(ctx);
            if (i === imgList.length - 1) {
                encoder.finish();
                next()
            }
        }
    });
```
## Error output

```js
App 3464715 output: node:internal/modules/cjs/loader:1280
App 3464715 output:   return process.dlopen(module, path.toNamespacedPath(filename));
App 3464715 output:                  ^
App 3464715 output: Error: /<path>/modules/canvas/build/Release/canvas.node: invalid ELF header
App 3464715 output:     at Object.Module._extensions..node (node:internal/modules/cjs/loader:1280:18)
App 3464715 output:     at Module.load (node:internal/modules/cjs/loader:1074:32)
App 3464715 output:     at Function.Module._load (node:internal/modules/cjs/loader:909:12)
App 3464715 output:     at Module.require (node:internal/modules/cjs/loader:1098:19)
App 3464715 output:     at Module.require (/opt/cpanel/ea-ruby27/root/usr/share/passenger/helper-scripts/node-loader.js:80:25)
App 3464715 output:     at require (node:internal/modules/cjs/helpers:108:18)
App 3464715 output:     at Object.<anonymous> (/<path>/modules/canvas/lib/bindings.js:3:18)
App 3464715 output:     at Module._compile (node:internal/modules/cjs/loader:1196:14)
App 3464715 output:     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1250:10)
App 3464715 output:     at Module.load (node:internal/modules/cjs/loader:1074:32) {
App 3464715 output:   code: 'ERR_DLOPEN_FAILED'
App 3464715 output: }
```

## Your Environment
* Version of node-canvas (output of `npm list canvas@2.11.2`):
* Environment (e.g. node 16.20.0 on Linux):

**Comment by zbjornson** [comment]
That usually means canvas was installed for a different platform. Please see the troubleshooting guide (#1511) and previous issues for this error: https://github.com/Automattic/node-canvas/issues?q=is%3Aissue+invalid+elf+header

**Comment by sarwanand24** [comment]
hey did you got the solution

**Comment by Whobeu** [comment]
I posted this for an open issue at [https://github.com/automattic/node-canvas/issues/2528](https://github.com/automattic/node-canvas/issues/2528):

Just ran into this with a new Pi 5 4GB that I am testing with a Raspberry Pi Bookworm image. Raspberry Pi forums suggest putting "kernel=kernel8.img" in the /boot/firmware/config.txt file (I put it in the [all] section though a specific [pi5] section would work too). Tried it and node-canvas works now but having other issues unrelated to node-canvas. YMMV

UPDATE: I decided to dig in a little further as I felt that using the native kernel_2712.img would be better for performance and also expose various kernel specific options for the Pi 5. I had installed the Cairo libraries using apt but it appeared they were not being used. Digging into the node_modules/canvas directory I found the build/Release directory contained all the Cairo libraries. On a hunch, I booted the Pi 5 kernel image kernel_2712.img and then removed all of the lib*.* modules from build/Release leaving only the canvas.node module. Bingo! The ELF error is gone and canvas is working fine. Of course the problem will be back should a reinstall of node_modules take place as the libraries will most likely return.

So there are basically two workarounds at the moments for the issue. Run kernel8.img or removed the Cairo libraries from node_modules and let the system library versions be used instead.
